server/test/views/components/TextFormatterTest.java

```diff
@@ -25,11 +26,48 @@ public void urlsRenderCorrectly() {
                 + " class=\"text-seattle-blue\">http://internet.website</a>");
   }
 
+  @Test
+  public void urlsRenderToOpenInNewTabCorrectly() {
+    ImmutableList<DomContent> content =
+        TextFormatter.createLinksAndEscapeText(
+            "hello google.com http://internet.website", TextFormatter.UrlOpenAction.NewTab);
+
+    assertThat(content).hasSize(4);
+    assertThat(content.get(0).render()).isEqualTo(new Text("hello ").render());
+    assertThat(content.get(1).render())
+        .isEqualTo(
+            "<a href=\"http://google.com/\" class=\"text-seattle-blue\" target=\"_blank\">google.com<svg xmlns=\"http://www.w3.org/2000/svg\" fill=\"currentColor\" stroke=\"currentColor\" stroke-width=\"1%\" aria-hidden=\"true\" viewBox=\"0 0 24 24\" width=\"24\" height=\"24\" class=\"flex-shrink-0 h-5 w-auto inline ml-1 align-text-top\"><path d=\"M19 19H5V5H12V3H5C3.89 3 3 3.9 3 5V19C3 20.1 3.89 21 5 21H19C20.1 21 21 20.1 21 19V12H19V19ZM14 3V5H17.59L7.76 14.83L9.17 16.24L19 6.41V10H21V3H14Z\"></path></svg></a>");
```

Comment:
Up to you, but I'd consider a less strict assertion to avoid having to inline the svg content (which may change at some point).

Maybe we could assert that the content starts with everything through `<svg ` and then assert that the content ends with `</svg></a>`. I know we lose some fidelity in what's being asserted with that approach, so understand if you'd prefer to keep the full SVG content inlined.

Same for the assertion below.

Reply:
Good idea. Fixed